packages/crypto/lib/models/transaction.js

```diff
@@ -223,28 +222,28 @@ module.exports = class Transaction {
       bb.append(keysgroupBuffer, 'hex')
     }
 
-    if (transaction.type === TRANSACTION_TYPES.IPFS) {
+    else if (transaction.type === TRANSACTION_TYPES.IPFS) {
       bb.writeByte(transaction.asset.ipfs.dag.length / 2)
       bb.append(transaction.asset.ipfs.dag, 'hex')
     }
 
-    if (transaction.type === TRANSACTION_TYPES.TIMELOCK_TRANSFER) {
+    else if (transaction.type === TRANSACTION_TYPES.TIMELOCK_TRANSFER) {
       bb.writeUInt64(+transaction.amount.toString())
       bb.writeByte(transaction.timelockType)
       bb.writeUInt32(transaction.timelock)
       bb.append(bs58check.decode(transaction.recipientId))
     }
 
-    if (transaction.type === TRANSACTION_TYPES.MULTI_PAYMENT) {
+    else if (transaction.type === TRANSACTION_TYPES.MULTI_PAYMENT) {
       bb.writeUInt32(transaction.asset.payments.length)
       transaction.asset.payments.forEach(p => {
         bb.writeUInt64(p.amount)
         bb.append(bs58check.decode(p.recipientId))
       })
     }
 
-    if (transaction.type === TRANSACTION_TYPES.DELEGATE_RESIGNATION) {
-      // delegate resignation - empty payload
+    else if (transaction.type === TRANSACTION_TYPES.DELEGATE_RESIGNATION) {
+      return
```

Comment:
What is the reason for returning here?

Reply:
says in previous comment that there is no payload to be serialized for delegate registration

Reply:
As @supaiku0 explained, the payload being empty only means that there is no asset to be processed but you still need to continue the processing otherwise.